🧑‍🌾 gz_launch-ci-gz-launch9-homebrew clang warning regarding variable lenght arrays in `WebsocketServer.cc` on mac sequoia

<!-- If you're not sure whether your problem is a bug, please ask a question at
https://robotics.stackexchange.com instead.-->

## Environment
* OS Version: mac sequoia
* Source

## Description
* Expected behavior: To build without warnings
* Actual behavior: Building with warnings

## Steps to reproduce
<!-- Provide steps so we can try to reproduce this issue -->

1. Go to https://build.osrfoundation.org/job/gz_launch-ci-gz-launch9-homebrew-arm64
2. Build a job
3. See the warnings

## Output
```
/Users/jenkins/jenkins-agent/workspace/gz_launch-ci-gz-launch9-homebrew-arm64/gz-launch/plugins/websocket_server/WebsocketServer.cc:177:27: warning: variable length arrays in C++ are a Clang extension [-Wvla-cxx-extension]
  177 |         unsigned char buf[buflen + LWS_PRE];
      |                           ^~~~~~~~~~~~~~~~
/Users/jenkins/jenkins-agent/workspace/gz_launch-ci-gz-launch9-homebrew-arm64/gz-launch/plugins/websocket_server/WebsocketServer.cc:177:27: note: read of non-const variable 'buflen' is not allowed in a constant expression
/Users/jenkins/jenkins-agent/workspace/gz_launch-ci-gz-launch9-homebrew-arm64/gz-launch/plugins/websocket_server/WebsocketServer.cc:176:16: note: declared here
  176 |         size_t buflen = strlen(format) + (conns.size() - 1);
      |                ^
1 warning generated.
```

CC: @Crola1702

--- a/plugins/websocket_server/WebsocketServer.cc
+++ b/plugins/websocket_server/WebsocketServer.cc
@@ -16,6 +16,7 @@
 */
 
 #include <algorithm>
+#include <vector>
 
 #include <gz/msgs/bytes.pb.h>
 #include <gz/msgs/empty.pb.h>
@@ -174,9 +175,9 @@ int httpCallback(struct lws *_wsi,
 
         // Prepare the output
         size_t buflen = strlen(format) + (conns.size() - 1);
-        unsigned char buf[buflen + LWS_PRE];
+        std::vector<unsigned char> buf(buflen + LWS_PRE);
         int n;
-        n = snprintf(reinterpret_cast<char *>(buf), buflen, format,
+        n = snprintf(reinterpret_cast<char *>(buf.data()), buflen, format,
             conns.c_str());
         // Check that no characters were discarded
         if (n - int(buflen) > 0)
@@ -192,8 +193,8 @@ int httpCallback(struct lws *_wsi,
 
         // Write response body
         lws_write_http(_wsi,
-                       reinterpret_cast<unsigned char *>(buf),
-                       strlen(reinterpret_cast<const char *>(buf)));
+                       buf.data(),
+                       strlen(reinterpret_cast<const char *>(buf.data())));
         break;
       }
       // Return a 404 if no route was matched

